applications/swarm-nl.md

```diff
@@ -0,0 +1,156 @@
+        * Selection and configuration of the transport layers to be supported by the node.
+        * Selection of the cryptographic keypair to use for identity generation e.g Edwards.
+        * Storage and retrieval of keypair locally.
+        * PeerID and multiaddress generation.
+        * Protocol specification and handlers.
+        * Event handlers for network events and logging.
+
+    * **Gossiping** -- SwarmNL will implement the [Gossipsub 1.1](https://github.com/libp2p/specs/blob/master/pubsub/gossipsub/gossipsub-v1.1.md) protocol, specified by the [libp2p spec](https://github.com/libp2p/specs). This would be easily configurable to accommodate varying and peculiar network conditions.
+        
+* **Scaling** -- SwarmNL needs to efficiently handle a growing (or shrinking) number of nodes while maintaining performance and reliability. Here's what we plan to implement to this effect:
+    * **Sharding** -- implementation of a flexible generic sharding protocol that allows application specify configurations like sharding hash functions and locations for shards.
+    * **Data Forwarding** -- definition of a protocol for forwarding messages between nodes in different shards and establishment of efficient routing mechanisms for inter-shard communication.
+    * **Fault Tolerance** -- implementation of fault-tolerant mechanisms for detecting (and recovering from) node failures. This might involve redundancy, node replication, erasure encoding/decoding or re-routing strategies. 
+    
+* **IPFS**:
+    * **Upload** -- provision of interfaces to upload to IPFS, pin on current node and post arbitrary data to remote servers. Encryption is also easily pluggable and will be provided.
+    * **Download** -- retrieval and possible decryption of data from the IPFS network.
+    
+
+#### Technology Stack
+
+* Libp2p
+* Rust
+
+
+#### What is not included in SwarmNL
+SwarmNL will not be involved in providing a storage solution. Because SwarmNL will be generic over the networking layer, it will be possible to easily plug in any storage interface in a flexible and configurable way.
+
+Storage maintained by the library will mostly be in memory with the exception of config files and other node state keeping data.
+
+### Ecosystem Fit
+- Where and how does your project fit into the ecosystem?
+    - SwarmNL can be used by applications or projects that require a bespoke decentralized communication layer for transporting and exchanging off-chain data.
+- Who is your target audience (parachain/dapp/wallet/UI developers, designers, your own user base, some dapp's userbase, yourself)?
+    - Our priority is to build this for [SamaritanDB](https://github.com/thewoodfish/samaritan-db). However, SwarmNL will be generic enough to be used by any developer in the ecosystem that would require a primitive networking layer
```

Comment:
Could you name some other good use cases where swarmNL could be leveraged?

Reply:
Thank you for your question. swarmNL could be leveraged in Content Delivery Networks (CDNs), distributed storage systems, IoT (Internet of Things), multiplayer online games and file hosting and sharing systems